internal/engine/build_and_deployer.go

```diff
@@ -87,34 +87,15 @@ func (composite *CompositeBuildAndDeployer) BuildAndDeploy(ctx context.Context,
 	return store.BuildResultSet{}, lastErr
 }
 
-func DefaultBuildOrder(sbad *SyncletBuildAndDeployer, cbad *LocalContainerBuildAndDeployer, ibad *ImageBuildAndDeployer, dcbad *DockerComposeBuildAndDeployer, env k8s.Env, updMode UpdateMode, runtime container.Runtime) BuildOrder {
-
+func DefaultBuildOrder(lubad *LiveUpdateBuildAndDeployer, ibad *ImageBuildAndDeployer, dcbad *DockerComposeBuildAndDeployer,
+	updMode UpdateMode, env k8s.Env, runtime container.Runtime) BuildOrder {
 	if updMode == UpdateModeImage || updMode == UpdateModeNaive {
 		return BuildOrder{dcbad, ibad}
 	}
 
-	if updMode == UpdateModeKubectlExec {
-		return BuildOrder{sbad, dcbad, ibad}
-	}
-
-	if updMode == UpdateModeContainer {
-		return BuildOrder{cbad, dcbad, ibad}
-	}
-
-	if updMode == UpdateModeSynclet {
-		if runtime == container.RuntimeDocker {
-			ibad.SetInjectSynclet(true)
-		}
-		return BuildOrder{sbad, dcbad, ibad}
-	}
-
-	if env.IsLocalCluster() && runtime == container.RuntimeDocker {
-		return BuildOrder{cbad, dcbad, ibad}
-	}
-
-	if runtime == container.RuntimeDocker {
+	if updMode == UpdateModeSynclet || (!env.IsLocalCluster() && runtime == container.RuntimeDocker) {
```

Comment:
i wonder if LiveUpdateBuildAndDeployer should have a canUseSynclet method, since it's important that the logic here and the logic in containerUpdaterForSpecs are in-sync, and right now it's hard to prove that they are